contributors/design-proposals/standardization-of-development-process.md

```diff
@@ -0,0 +1,282 @@
+# Proposal to Standardize Development Process
+
+## Links
+```
+---
+Links:
+- Issues:
+  - [someIssueURL]()
+- PRs:
+  - https://github.com/kubernetes/community/pull/967
+- MailingListDiscussions:
+  - https://groups.google.com/forum/#!topic/kubernetes-dev/65A-3ULYPB0
+- Documentation:
+  - [someDocsLinkURL]()
+```
+
+## Responsible SIG(s)
+
+- SIG Release
+- SIG PM
+- SIG Architecture
+- SIG Governance
+- Technical Steering Committee
+
+## Summary
+
+A standardized development process for Kubernetes is proposed in order to:
+- provide a common structure for proposing changes to Kubernetes
```

Comment:
A foundational concern for this proposal would be what constitutes a "change" that is in bounds for the process.  In the current workflow, I believe much of our struggle comes from arbitrary boundaries between features, enhancements, defect corrections, and technical debt elimination.  One person's bug fix is another person's feature.  Brian Grant and others have been fairly prescriptive about these definitions, but the implementation of a _proscriptive_ policy will force the issue.  Because open collaboration contributions usually follow the path of least resistance, it means that an increasing amount of work will bypass the process over time.  And, if the idea of the RFC process is to seed the entirety of feature and architecture backlogs, having it out of alignment with the actual code could present a serious risk.  In order to address this, extensive referential automation, documentation, and human attention may be required.  We will also need a very clear knife edge between all of the types of "changes" I mentioned above, as well as easy access to decision-making guidance.  If you study the pitfalls of implementing technical change management, you quickly learn how such processes can lead to disaster either through subversion or dropping velocity precipitously.  None of this is a disagreement with the premise of this PR, only illuminating known antipatterns.

Reply:
Practically, any work that significantly crosses SIG boundaries (and touches some special areas like APIs) will require coordination and this process.  As we move to associating code with SIGs, it'll become more more apparent when this happens.

Even though an individual person may have the authority to approve changes that cut across SIGs, we expect those folks to use their power wisely.  Instead of approving a big PR that hits a lot of areas, those are the folks that would suggest that a proposal be generated.

Reply:
@jbeda I feel that the production of release notes is a special case of work significantly crossing a SIG boundary. In my limited sampling of feature issues and pull requests there is a huge burden placed on our tech writers to explain changes in a format consumable by people besides the developers of a feature. There is also a large burden placed on the PM Group and the Release Team to actually determine what went into the release. in my opinion this is a huge risk for the future success of the project because we _need_ users to feel that it's important to keep their clusters up to date, however, if we can't produce a compelling narrative which encourages people to upgrade their critical infrastructure we're in bad shape as a project